### DISCUSSION TITLE
Ui suggestion

### DISCUSSION THREAD

**Discussion by bastiencyr** [borgbase/vorta#993] [Ideas]
Hello, 
I checked the mainWindow ui file because I wanted to optimize the space at the bottom (on my screen, the window is to big the first time). There is a status bar and and it seems that noboy use it so I have removed it. I have also moved the cancel and backup buttons to a toolBar. 
This is the result :
![image](https://user-images.githubusercontent.com/63359575/118247637-b12b3580-b4a3-11eb-9c90-9a51e88b43a6.png)

It's just a suggestion but I think a toolBar is a good idea. It is more compact and beautiful (with icons). 
My "Backup" button on toolBar starts a backup and it works.

I have others suggestions like adding the misc tab in a setting action in toolBar (all ui is made for one profile, one backup but misc tab is about all profiles and backups,  its global settings), keep misc tab for repository settings and add compression option in it, maybe move the profile menu elsewhere...

**Comment by m3nu** [comment]
Agree that the status bar could be removed or used for something.

Moving buttons to the menu might be a bit unexpected for users.

**Comment by bastiencyr** [comment]
I was thinking for example of Timeshift, it's clearer with a big button.  
Is it possible to integrate this project to propose and implement other ideas ?

**Comment by m3nu** [comment]
You mean like this screenshot? Pull requests that improve Vorta are always welcome. Just discuss your ideas before implementing. And dont copy code directly from other projects. Taking ideas and inspirations is ok though. 👌

![](https://res.cloudinary.com/practicaldev/image/fetch/s--jQ1drHaj--/c_limit%2Cf_auto%2Cfl_progressive%2Cq_auto%2Cw_880/https://dev-to-uploads.s3.amazonaws.com/i/wif9rs60am8l5lg01fy3.png)

**Comment by bastiencyr** [comment]
Yes just like this. It's just inspiration. 
So yes, you have a toolbar with the most important options. 

I have other idea. I started to implement this. Look at this : 
So this is vorta without modification : 
![image](https://user-images.githubusercontent.com/63359575/118351249-2b6ac100-b55b-11eb-90db-541ddc0c70ae.png)

Actually, there are two modifications : the status bar has been removed and I added the toolbar. We already have talked about that. But notice that box are spaced. I think that the interface is is not sufficiently spaced. So look at this new suggestion : 
![vorta](https://user-images.githubusercontent.com/63359575/119359229-b4939d80-bca9-11eb-92d0-447b05a4b93d.png)


Maybe you see my new source of inspiration hh (settings of keepassXC). It was not easy to do this. On the left, it's the repository, sources, clock, archives and misc tabs.   

It's the main modification but I have added many others minor modification.  Notice that on Linux, it seems to be more compact that on Mac without modifications. The tab organised vertically (for clock, wifi, tasks, terminal for example) makes these tabs hard to see
I reduced the space at the bottom, so the text takes only two spaces in heights instead of 3. I added spaces In misc, for wifi items,  clock, repository.
The text at the bottom is centered but maybe its better to put this text on the right. I cant put the text on the left, its not aesthetic with the vertical tab bar.

Maybe I should move this issue to Discussions

**Comment by bastiencyr** [comment]
@m3nu ?

**Reply by m3nu** [reply]
Moved to a discussion.

We can't move *Start Backup/Cancel Backup* to be a menu because such a UI would be highly unusual and it's not a menu item at all.

Making the tabs vertical is possible, but seems to need more space and I don't see any benefit from changing it.

The main feedback we got UI-wise is that it can be difficult to get started. Putting the repo tab first tries to solve this in a way. But could be improved.

Then later, I mostly work in the *Source* and *Archive* tabs.

These are the improvements that could make a difference from my POV.

**Reply by bastiencyr** [reply]
> We can't move _Start Backup/Cancel Backup_ to be a menu because such a UI would be highly unusual and it's not a menu item at all.

It's not a menu but buttons in toolbar. 

> Making the tabs vertical is possible, but seems to need more space and I don't see any benefit from changing it.

Actually, it doesnt take more space. The vorta ui, like all ui, takes spaces vertically and doesn't exploit horizontal space.
So, on my little screen (12,5 inch), I don't feel like I'm losing space (with my proposal) , more like gaining more space because laptop screens have more horizontal space to use than vertical one (rectangle form factor).

It's clearer like that. The horizontal tabs like clock/wifi/tasks/terminal are not obvious. 

Finally, I find it prettier but its a matter of taste, you should ask more people. Note that most modern UI's have a side bar for better readability.

**Comment by m3nu** [comment]
> It's clearer like that. The horizontal tabs like clock/wifi/tasks/terminal are not obvious.

The window you posted is much larger than the current default size. So the horizontal tabs will use about 15% more space on one side?

Regarding the Start/Cancel buttons, moving them will detach them from the logs that belong with it. So I'm not a fan of that. And the unusual placement of buttons where one usually finds the menu.

Here some other UI concepts to explore. Similar complexity. The icon tabs on top are a common theme.

![Screen Shot 2021-05-25 at 13 26 17](https://user-images.githubusercontent.com/3916435/119474316-13661f00-bd5d-11eb-8213-d7fdb2fa1ad7.jpg)

![Screen Shot 2021-05-25 at 13 26 11](https://user-images.githubusercontent.com/3916435/119474301-0fd29800-bd5d-11eb-94ee-5b4c64ba941b.jpg)

![Screen Shot 2021-05-25 at 13 43 42](https://user-images.githubusercontent.com/3916435/119476712-5d500480-bd5f-11eb-9609-e916523842c1.jpg)

![Screen Shot 2021-05-25 at 13 44 32](https://user-images.githubusercontent.com/3916435/119476724-604af500-bd5f-11eb-9e60-2c637e10f7d4.jpg)

![Screen Shot 2021-05-25 at 13 46 34](https://user-images.githubusercontent.com/3916435/119477005-a902ae00-bd5f-11eb-9226-a04e16a0dadf.jpg)

**Comment by bastiencyr** [comment]
>  The window you posted is much larger than the current default size. So the horizontal tabs will use about 15% more space on one side?

I just checked, its the same factor between width and height and width is the same in centimetre (I have taken a rule).

> Regarding the Start/Cancel buttons, moving them will detach them from the logs that belong with it. So I'm not a fan of that. And the unusual placement of buttons where one usually finds the menu.

True true, you're right. I don't think it's such a good idea

> Here some other UI concepts to explore. Similar complexity. The icon tabs on top are a common theme.

I like the first two. Maybe I can do something configurable ? User can choose between two interfaces.

**Reply by m3nu** [reply]
Definitely not maintaining 2 interfaces. 😜 If we come up with a new idea that has significant benefits over the current one, we add it. Else not.

**Reply by bastiencyr** [reply]
hhh so which one ? My preference is for the second (without the cell of course). But what about you ?

**Reply by m3nu** [reply]
The battery settings? Yeah thats nice. So we would have a profile switcher somewhere, then icons on the left and actual settings with tabs on the right. And logs at the bottom?

**Reply by bastiencyr** [reply]
So, for the moment, I will try to move tabs on the left with icons just like the first UI with the cell. We will see better where to put the profile switcher. 
I dont understand what do you mean by moving "actual settings with tabs on the right" ? On the first UI there isnt settings with tabs on the right.

**Reply by m3nu** [reply]
> I dont understand what do you mean by moving "actual settings with tabs on the right" ? On the first UI there isnt settings with tabs on the right.

I meant that we can use normal tabs instead of the QToolbox you don't like.

![pyqt-toolbox](https://user-images.githubusercontent.com/3916435/119521091-e3824000-bd8b-11eb-8f87-2ad435ba9b38.png)

![Screen Shot 2021-05-25 at 19 01 48](https://user-images.githubusercontent.com/3916435/119521111-e715c700-bd8b-11eb-9ddb-24441c772dfb.jpg)

**Reply by bastiencyr** [reply]
oooh I get it, yes I am not a big fan of QToolbox. Fine. I will try to implement that (it's not the same way that the first idea), and I will come back when I have a new interface proposal similar to the first image.  Thanks for you reply and patience 😄😄

**Reply by m3nu** [reply]
Best to wait until #955 is merged, since it also edits some of the UI files. Just to avoid merge conflicts.

**Comment by bastiencyr** [comment]
> Best to wait until #955 is merged, since it also edits some of the UI files. Just to avoid merge conflicts.

No problem. I have to do some tests to be sure. 
For the moment, I can propose this : 

https://user-images.githubusercontent.com/63359575/119719026-958f3a00-be68-11eb-9bd6-6464db14f73a.mp4

The text is at the bottom just like before. And profile on the left. But I dont think is a good idea. Just put the profile ComboBox at the top like before is ok.

**Comment by bastiencyr** [comment]
Hello, @m3nu , can I make a pull request since #955 have been merging ?

**Reply by m3nu** [reply]
Since it's the UI, please let's plan and sketch/draft it out first. I don't want you to spend time on code that's maybe not suitable to merge later. It will also be a lot of work to get all the details right. Easy to underestimate.

If you're ready for that, then let's see what you have.

**Reply by bastiencyr** [reply]
I think its ready. It was not big edits over code. I rebased without conflict ! I push :)
